akka-docs/rst/java/code/docs/http/javadsl/server/directives/FileAndResourceDirectivesExamplesTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2016-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package docs.http.javadsl.server.directives;
+
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.server.PathMatchers;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.server.directives.DirectoryRenderer;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import org.junit.Ignore;
+import org.junit.Test;
+import scala.NotImplementedError;
+
+import static akka.http.javadsl.server.PathMatchers.segment;
+
+public class FileAndResourceDirectivesExamplesTest extends JUnitRouteTest {
+
+  @Ignore("Compile only test")
+  @Test
+  public void testGetFromFile() {
+    //#getFromFile
+    final Route route = path(PathMatchers.segment("logs").slash(segment()), name ->
+      getFromFile(".log")
+    );
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/logs/example"))
+      .assertEntity("example file contents");
+    //#getFromFile
+  }
+
+  @Ignore("Compile only test")
+  @Test
+  public void testGetFromResource() {
+    //#getFromResource
+    final Route route = path(PathMatchers.segment("logs").slash(segment()), name ->
+      getFromResource(".log")
+    );
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/logs/example"))
+      .assertEntity("example file contents");
+    //#getFromResource
+  }
+
+  @Ignore("Compile only test")
+  @Test
+  public void testListDirectoryContents() {
+    //#listDirectoryContents
+    final Route route = route(
+      path("tmp", () -> listDirectoryContents("/tmp")),
+      path("custom", () -> {
+        final DirectoryRenderer renderer = renderVanityFooter -> {
+          throw new NotImplementedError();
+        };
```

Comment:
best case we would actually have a short implementation both here and in the scala test, but let's just make that a separate ticket